src/main/java/com/emc/ecs/sync/storage/azure/AzureBlobStorage.java

```diff
@@ -0,0 +1,341 @@
+
+        } catch (IllegalArgumentException | URISyntaxException | InvalidKeyException | StorageException e) {
+            throw new ConfigurationException("error connecting to Azure blob storage account: " + e);
+        }
+    }
+
+    @Override
+    public String getRelativePath(String identifier, boolean directory) {
+        return identifier;
+    }
+
+    @Override
+    public String getIdentifier(String relativePath, boolean directory) {
+        return relativePath;
+    }
+
+    @Override
+    protected ObjectSummary createSummary(final String identifier) {
+        BlobProperties blobProperties = getCloudBlobReference(identifier).getProperties();
+        return new ObjectSummary(identifier, false, blobProperties.getLength());
+    }
+
+    @Override
+    public Iterable<ObjectSummary> allObjects() {
+        return () -> new PrefixIterator(config.getBlobPerfix());
+    }
+
+    private class PrefixIterator extends ReadOnlyIterator<ObjectSummary> {
+        private String prefix;
+        private List<ObjectSummary> objectSummaries = new ArrayList<>();
+        private Iterator<ObjectSummary> itemIterator;
+
+        PrefixIterator(String prefix) {
+            this.prefix = prefix;
+            getItemIterator();
+        }
+
+        private void getItemIterator() {
+            Iterator<ListBlobItem> blobItemIterator = time(() -> container.listBlobs(prefix).iterator(), OPERATION_LIST_BLOBS);
+            while (blobItemIterator.hasNext()) {
+                listAllBlobSummary(objectSummaries, blobItemIterator.next(), prefix);
+            }
+
+            log.info("total objectSummaies:{}", objectSummaries.size());
+            itemIterator = objectSummaries.iterator();
+        }
+
+        @Override
+        protected ObjectSummary getNextObject() {
+            if (itemIterator.hasNext()) {
+                return itemIterator.next();
+            } else {
+                return null;
+            }
+        }
+
+        private void listAllBlobSummary(List<ObjectSummary> objectSummaries, ListBlobItem blob, String perfix) {
+            if (blob instanceof CloudBlobDirectory) {
+                try {
+                    for (ListBlobItem blobItem : ((CloudBlobDirectory) blob).listBlobs(perfix)) {
+                        listAllBlobSummary(objectSummaries, blobItem, perfix);
+                    }
+                } catch (URISyntaxException | StorageException e) {
+                    throw new RuntimeException("error for list all Azure blobs");
+                }
+            } else if (blob instanceof CloudBlob) {
+                objectSummaries.add(new ObjectSummary(((CloudBlob) blob).getName(), false, ((CloudBlob) blob).getProperties().getLength()));
+            } else {
+                throw new RuntimeException("wrong blob type: " + blob.getUri());
+            }
+        }
+    }
+
+    //use PrefixIterator to get all objects
+    private class PrefixWithSnapshotIterator extends ReadOnlyIterator<ObjectSummary> {
```

Comment:
there should be no difference in object enumeration whether you're including snapshots or not (see below)